### PR TITLE
Larvitar async

### PR DESCRIPTION
**Pull request by daron1337**
Added promises instead of callbacks.

See Migration.md for breaking changes w/ old versions

### CODE REVIEW

File: imaging/image_parsing.js [hunk 1]
```diff
@@ -123,9 +126,11 @@ export const dumpDataSet = function (dataSet, metadata, customFilter) {
  * @function parseNextFile
  * @param {Array} parsingQueue - Array of queued files to be parsed
  * @param {Object} allSeriesStack - Series stack object to be populated
- * @param {Function} callback - Passed through
+ * @param {Function} resolve - Promise resolve function
+ * @param {Function} reject - Promise reject function
+ * @returns {Promise} - Return a promise which will resolve to a image object list or fail if an error occurs
```
Comment (akiross) [inline imaging/image_parsing.js:131]: This is not returning a Promise, remove this line.

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Introduction
+
+This guide is primarily for user with prior Larvitar v0.x experience who want to learn about the new features and changes in Larvitar v1.x. While it looks like a lot has changed, a lot of what you know and love abut Larvitar is still the same; bit we wanted to be as thorough as possibile and provide detailed explanations or examples for every documented change.
```
Comment (akiross) [inline MIGRATION.md:3]: typo: BUT we wanted to be as...

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
     "dicom",
     "imaging"
   ],
-  "version": "0.20.0",
+  "version": "1.0.0",
```
Comment (akiross) [inline package.json:7]: Be sure to include `BREAKING CHANGE: blah` (or use `!` e.g. `fix!: blah`, `feat!: blah`) in commit message.

**Review by akiross — CHANGES_REQUESTED** [review]
It could be a good chance to also fix a couple of minor inconsistencies:
  - loaders are named `multiframeLoader.js` and `nrrdLoader.js` using camel case, instead all the other files are using `snake_case.js`. It would be nice to rename those files as well.
  - `tools/tools.segmentation.js` have a redundant `tool` in the name, can it be removed?

**Review by akiross — APPROVED** [review]
r+

**Comment by ronzim** [comment]
Custom tools files were camelCase to be consistent with the original cs tools filename: https://github.com/cornerstonejs/cornerstoneTools/tree/master/src/tools

File: imaging/image_tools.js [hunk 4]
```diff
@@ -272,17 +268,13 @@ export const clearMeasurements = function (cb) {
   each(enabledElements, el => {
     cornerstone.updateImage(el.element);
   });
-
-  if (cb) {
-    cb();
-  }
 };
```
Comment (ronzim) [inline imaging/image_tools.js:271]: Why no callback & no promise here ? 🤔

**Comment by daron1337** [comment]
> Custom tools files were camelCase to be consistent with the original cs tools filename: https://github.com/cornerstonejs/cornerstoneTools/tree/master/src/tools

Ouch! @akiross ?

File: imaging/image_tools.js (hunk 4 above)
Comment (daron1337) [reply imaging/image_tools.js:271]: It's a sync function

File: MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Introduction
+
+This guide is primarily for user with prior Larvitar v0.x experience who want to learn about the new features and changes in Larvitar v1.x. While it looks like a lot has changed, a lot of what you know and love abut Larvitar is still the same; but we wanted to be as thorough as possibile and provide detailed explanations or examples for every documented change.
```
Comment (ronzim) [inline MIGRATION.md:3]: Typo "about"

**Review by ronzim — APPROVED** [review]

**Pull request merged**